examples/src/big_compute_buffers/mod.rs

```
@@ -0,0 +1,267 @@
+use std::{borrow::Cow, num::NonZeroU32};
+use wgpu::{util::DeviceExt, BufferSlice, Features};
+
+// These are set by the minimum required defaults for webgpu.
+const MAX_BUFFER_SIZE: u64 = 1 << 27; // 134_217_728 // 134MB
+const MAX_DISPATCH_SIZE: u32 = (1 << 16) - 1;
+
+pub async fn execute_gpu(numbers: &[f32]) -> Vec<f32> {
+    let instance = wgpu::Instance::default();
+
+    let adapter = instance
+        .request_adapter(&wgpu::RequestAdapterOptions::default())
+        .await
+        .unwrap();
+
+    let (device, queue) = adapter
+        .request_device(
+            &wgpu::DeviceDescriptor {
+                label: None,
+                // These features are required to use `binding_array` in your wgsl.
+                // Without them your shader may fail to compile.
+                required_features: Features::STORAGE_RESOURCE_BINDING_ARRAY
+                    | Features::BUFFER_BINDING_ARRAY
+                    | Features::UNIFORM_BUFFER_AND_STORAGE_TEXTURE_ARRAY_NON_UNIFORM_INDEXING,
+
+                memory_hints: wgpu::MemoryHints::Performance,
+                required_limits: wgpu::Limits {
+                    max_buffer_size: MAX_BUFFER_SIZE,
+                    ..Default::default()
+                },
+            },
+            None,
+        )
+        .await
+        .unwrap();
+
+    execute_gpu_inner(&device, &queue, numbers).await.unwrap()
+}
+
+pub async fn execute_gpu_inner(
+    device: &wgpu::Device,
+    queue: &wgpu::Queue,
+    numbers: &[f32],
+) -> Option<Vec<f32>> {
```

This, too, should just return a `Vec<f32>`, without the `Option`. Instead of logging a message and returning `None` if there's an error, it should just use `panic!` to report that message.